backends/gstreamer/lib.rs

```diff
@@ -58,36 +59,50 @@ use servo_media_player::{Player, PlayerEvent, StreamType};
 use servo_media_streams::capture::MediaTrackConstraintSet;
 use servo_media_streams::registry::MediaStreamId;
 use servo_media_streams::MediaOutput;
+use servo_media_traits::muteable::Muteable;
+use servo_media_traits::ClientContextId;
 use servo_media_webrtc::{WebRtcBackend, WebRtcController, WebRtcSignaller};
+use std::collections::HashMap;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, Mutex};
+use std::vec::Vec;
 
 lazy_static! {
     static ref BACKEND_BASE_TIME: gst::ClockTime = { gst::SystemClock::obtain().get_time() };
 }
 
 pub struct GStreamerBackend {
     capture_mocking: AtomicBool,
+    muteables: Mutex<HashMap<ClientContextId, Vec<Arc<Mutex<dyn Muteable>>>>>,
```

Comment:
We need a way to remove `Muteable`s from this map when they are dropped.

Here's an idea to fix this:

- Add `shutdown_player(&self, client_context_id: &ClientContextId, player: Arc<Mutex<dyn Player>)` and `shutdown_audio_context(&self, client_context_id: &ClientContextId, audio_context: Arc<Mutex<AudioContext>>)` methods to the `Backend` trait.
- For the GStreamer implementation of these methods, remove the `Player` or `AudioContext` being shutdown from the `muteables` map. Note that if the `Vec` is empty after removing the `muteable`, we should remove the client context entry as well. We should also call the `Player.shutdown()` method within the `shutdown_player` implementation.
- To make sure that we don't keep expired instances alive if the client forgets to call these `shutdown_*` methods, one thing that we can do here is to use `Weak` instead of `Arc`. So: `muteables: Mutex<HashMap<ClientContextId, Vec<Weak<Mutex<dyn Muteable>>>>>,`.